### ISSUE TITLE
test_amici_objective fails repeatedly due to tolerances

### ISSUE THREAD

**Issue by yannikschaelte** [ICB-DCM/pyPESTO#276]
Even after relaxing ATOL+RTOL requirements, the test_preeq__guesses test fails roughly every 4th time. Any idea how to make this more stable?

Typical outcome:

```
def test_preeq_guesses():
        """
        Test whether optimization with preequilibration guesses works, asserts
        that steadystate guesses are written and checks that gradient is still
        correct with guesses set
        """
        petab_problem = petab.Problem.from_yaml(
            folder_base + "Zheng_PNAS2012/Zheng_PNAS2012.yaml")
        petab_problem.model_name = "Zheng_PNAS2012"
        importer = pypesto.PetabImporter(petab_problem)
        obj = importer.create_objective()
        problem = importer.create_problem(obj)
        optimizer = pypesto.ScipyOptimizer('ls_trf')
    
        result = pypesto.minimize(
            problem=problem, optimizer=optimizer, n_starts=2,
        )
    
        assert problem.objective.steadystate_guesses['fval'] < np.inf
        assert len(obj.steadystate_guesses['data']) == 1
    
        df = obj.check_grad(
            result.optimize_result.list[0]['x'],
            eps=1e-3,
            verbosity=0,
            mode=pypesto.objective.constants.MODE_FUN
        )
        print("relative errors MODE_FUN: ", df.rel_err.values)
        print("absolute errors MODE_FUN: ", df.abs_err.values)
>       assert np.all((df.rel_err.values < RTOL) | (df.abs_err.values < ATOL))
E       assert False
E        +  where False = <function all at 0x7fe67c1342f0>((array([5.54172708e-01, 1.44246439e+00, 4.23291604e-01, 1.16858930e+00,\n       1.23048130e-02, 9.63052330e-03, 6.489470...103e-01,\n       4.96319455e-01, 1.33405439e+00, 8.19742622e-01, 7.44499599e-02,\n       4.66051048e-02, 3.57963040e-06]) < 1e-05 | array([2.41588429e-02, 2.81097880e-02, 1.68579204e-03, 5.72546587e-04,\n       1.24040727e-03, 8.42903087e-04, 4.199184...820e-03,\n       1.09074214e-03, 3.57366666e-03, 4.53864977e-03, 1.17451909e-04,\n       4.85520242e-05, 3.18947311e-04]) < 1e-05))
E        +    where <function all at 0x7fe67c1342f0> = np.all
E        +    and   array([5.54172708e-01, 1.44246439e+00, 4.23291604e-01, 1.16858930e+00,\n       1.23048130e-02, 9.63052330e-03, 6.489470...103e-01,\n       4.96319455e-01, 1.33405439e+00, 8.19742622e-01, 7.44499599e-02,\n       4.66051048e-02, 3.57963040e-06]) = 0       0.554173\n1       1.442464\n2       0.423292\n3       1.168589\n4       0.012305\n5       0.009631\n6       0.006489...6319\n41      1.334054\n42      0.819743\n43      0.074450\n44      0.046605\n45      0.000004\nName: rel_err, dtype: float64.values
E        +      where 0       0.554173\n1       1.442464\n2       0.423292\n3       1.168589\n4       0.012305\n5       0.009631\n6       0.006489...6319\n41      1.334054\n42      0.819743\n43      0.074450\n44      0.046605\n45      0.000004\nName: rel_err, dtype: float64 =          grad       fd_f       fd_b       fd_c     fd_err   abs_err     rel_err\n0    0.018436  16.007333 -15.922144   ...000042   0.011759  0.000049    0.046605\n45  89.099945  89.244188  88.955065  89.099626   0.289123  0.000319    0.000004.rel_err
E        +    and   array([2.41588429e-02, 2.81097880e-02, 1.68579204e-03, 5.72546587e-04,\n       1.24040727e-03, 8.42903087e-04, 4.199184...820e-03,\n       1.09074214e-03, 3.57366666e-03, 4.53864977e-03, 1.17451909e-04,\n       4.85520242e-05, 3.18947311e-04]) = 0     0.024159\n1     0.028110\n2     0.001686\n3     0.000573\n4     0.001240\n5     0.000843\n6     0.000420\n7     0.01305...40    0.001091\n41    0.003574\n42    0.004539\n43    0.000117\n44    0.000049\n45    0.000319\nName: abs_err, dtype: float64.values
E        +      where 0     0.024159\n1     0.028110\n2     0.001686\n3     0.000573\n4     0.001240\n5     0.000843\n6     0.000420\n7     0.01305...40    0.001091\n41    0.003574\n42    0.004539\n43    0.000117\n44    0.000049\n45    0.000319\nName: abs_err, dtype: float64 =          grad       fd_f       fd_b       fd_c     fd_err   abs_err     rel_err\n0    0.018436  16.007333 -15.922144   ...000042   0.011759  0.000049    0.046605\n45  89.099945  89.244188  88.955065  89.099626   0.289123  0.000319    0.000004.abs_err
```

**Comment by FFroehlich** [comment]
change integration tolerances and play with FD step-size (eps)? Not too familiar with the Zheng model.

**Comment by yannikschaelte** [comment]
fixed in #305
